xbmc/epg/EpgContainer.cpp

```diff
@@ -235,20 +235,16 @@ void CEpgContainer::LoadFromDB(void)
 
 bool CEpgContainer::PersistTables(void)
 {
-  m_critSection.lock();
-  std::map<unsigned int, CEpg*> copy = m_epgs;
-  m_critSection.unlock();
-  return m_database.Persist(copy);
+  CSingleLock lock(m_critSection);
+  return m_database.Persist(m_epgs);
 }
 
 bool CEpgContainer::PersistAll(void)
 {
   bool bReturn(true);
-  m_critSection.lock();
-  std::map<unsigned int, CEpg*> copy = m_epgs;
-  m_critSection.unlock();
+  CSingleLock lock(m_critSection);
 
-  for (EPGMAP_CITR it = copy.begin(); it != copy.end() && !m_bStop; it++)
+  for (EPGMAP_CITR it = m_epgs.begin(); it != m_epgs.end() && !m_bStop; it++)
```

Comment:
My personal meaning for this is: The database HAS to be read befor the client search, because the client is not looking into the database but in the internal cache. This means that the cache must be filled before clients are added.
I hat 1600 EPG channels after a day in my database, because the client scan doesnt find the channels wich were currently in loading process. 1500 of the 1600 EPG channels were dead.